pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -244,24 +251,34 @@ func computeActiveCondition(ctx context.Context, pa *pav1alpha1.PodAutoscaler, p
 
 	switch {
 	case pc.want == 0:
-		if pa.Status.IsActivating() {
+		if pa.Status.IsActivating() && minReady > 0 {
 			// We only ever scale to zero while activating if we fail to activate within the progress deadline.
 			pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
 		} else {
 			pa.Status.MarkInactive("NoTraffic", "The target is not receiving traffic.")
 		}
+		// This happens in the initialScale 0 case. After SKS has been setup, we override want -1
+		// with 0 when PA's desired scale is not set yet.
+		if pc.ready >= minReady {
+			pa.Status.MarkScaleTargetInitialized()
+		}
 
 	case pc.ready < minReady:
 		if pc.want > 0 || !pa.Status.IsInactive() {
 			pa.Status.MarkActivating(
 				"Queued", "Requests to the target are being buffered as resources are provisioned.")
+		} else {
+			// Need to set a condition because otherwise we will get NewObservedGenFailure because the reconciler
+			// does not set any condition during reconciliation of a new generation
+			pa.Status.MarkInactive("NoTraffic", "The target is not receiving traffic.")
```

Comment:
How do we end up here? Since your change should only pass through `pc.want == 0` case?

Reply:
Do do hit this after minReady switches from 0 to 1. Even though PA has been marked inactive before in the pc.want==0 path, we'll get the `NewObservedGenFailure` if this is not set again.

Reply:
hm... so we set minReady to 0, then mark initial scale achieved to true, which means minReady becomes 1 on the next iteration, right?
But even on the next iteration pc.want should be 0 and go into the first case statement? Or it is -1 now and that's why we end up here?

Reply:
Yes exactly, pc.want is 0 for only one iteration. As soon as we mark ScaleTargetInitialized, it becomes -1 again.